lib/fiddle/struct.rb

```diff
@@ -261,21 +299,24 @@ def []=(*args)
       end
     end
 
+    def size=(new_size) # :nodoc:
+      raise ArgumentError, 'cannot change the size of a struct'
+    end
+
```

Comment:
How about using `undef_method`?

```suggestion
    undef_method :size=
```